seed/seedwriter/writer.go

```diff
@@ -291,9 +320,18 @@ func (w *Writer) SetOptionsSnaps(optSnaps []*OptionSnap) error {
 				return fmt.Errorf("snap %q is repeated in options", snapName)
 			}
 			w.byNameOptSnaps.Add(sn)
+		} else {
+			if !strings.HasSuffix(sn.Path, ".snap") {
```

Comment:
Nitpick and not for this PR, but we might think of having a simple helper for this somewhere around `snap/` (there are a couple of checks like this in the code).